noir/bootstrap.sh

```diff
@@ -15,8 +15,11 @@ if [ -n "$CMD" ]; then
   fi
 fi
 
-# Attempt to just pull artefacts from CI and exit on success.
-[ -n "${USE_CACHE:-}" ] && ./bootstrap_cache.sh && exit
+# Attempt to pull artifacts from CI if USE_CACHE is set and verify nargo usability.
+if [ -n "${USE_CACHE:-}" ]; then
+    ./bootstrap_cache.sh && ./noir-repo/target/release/nargo --version >/dev/null 2>&1 && exit 0
```

Comment:
Would it make sense to run nargo pulled from cache with rosetta until we have an arm image in teh cache?

Reply:
I'm certainly not opposed, but it is not immediately clear to me how to do that. Feel free to modify this PR!